### PR TITLE
[CBRD-24628] Change the csql CLI library not to use ncurses in the Linux.

### PR DESCRIPTION
**Pull request by eido5**
http://jira.cubrid.org/browse/CBRD-24628

**Purpose**

On the csql execution, when the user type the 'tab' key, the csql shows the current directory elements.(directories, files, etc)
To user, it seems unexpected execution. So, it should not be executed.
We try to eliminate this function, and then the CLI(command-line interface) should be changed that the reason of unexpected execution occurred from typing 'tab' key.
Without tab key, every function must be worked.

The entry of CLI library is below;
[simple editline](https://github.com/troglobit/editline#api)

**Implementation**

el_prev_hist() : return the ptr of history string from the latest one.
el_next_hist () :  return the ptr of history string from the oldest one.
csql_edit_contents_append() : store a history entry in the edit buffer.
add_history() : add a history in the CLi library buffer

**Remarks**

N/A

### CODE REVIEW

**Comment by airnet73** [comment]
A build error has occurred. Please check.

**Comment by eido5** [comment]
> 

Thank you I will do rebase onto this pr again.

**Comment by kisoo-han** [comment]
@eido5 ,
Would you describe following in description:
1. Remove ncurses dependency from csql
2. change of history related library from libedit to editline

**Comment by eido5** [comment]
Build test will not pass because I must change the path of CLi.
It will be fixed soon. (about 1 or 2 days)
I should create other repo and update modified CLi. So, It takes times a little bit.

@kisoo-han 
Could we ignore for a while till CLi will be modified?

**Review by mhoh3963 — COMMENTED** [review]
I suggest a few thing.

(1) The code complexity seems to have increased.
      so, how about making history_get() with el_prev_hist() and wl_next_hist() ?
     
(2) I knew that the replaced library support saving and reading a history string in the file.
      How about adding  new feature to read and save in the history file using this function ?

File: src/executables/csql_support.c [hunk 1]
```diff
@@ -1336,3 +1337,11 @@ csql_errmsg (int code)
       return (csql_get_message (CSQL_E_UNKNOWN_TEXT));
     }
 }
+
```
Comment (kisoo-han) [inline src/executables/csql_support.c:1340]: #if !defined(WINDOWS)

File: src/executables/csql_support.c [hunk 2]
```diff
@@ -1336,3 +1337,11 @@ csql_errmsg (int code)
...
+void
+stifle_history (int max_history_size)
+{
+  el_hist_size = max_history_size;
+
+  return;
+}
```
Comment (kisoo-han) [inline src/executables/csql_support.c]:
```suggestion
#if !defined(WINDOWS)
void
stifle_history (int max_history_size)
{
  el_hist_size = max_history_size;

  return;
}
#endif /* !WINDOWS */
```

Comment (eido5) [reply src/executables/csql_support.c]: I will apply it. 

Thank you.

File: src/executables/csql.c [hunk 3]
```diff
@@ -1806,6 +1842,13 @@ csql_execute_statements (const CSQL_ARGUMENT * csql_arg, int type, const void *s
       logddl_set_csql_input_type (CSQL_INPUT_TYPE_EDITOR);
     }
 
+  strcpy (prompt_cpy, stmts);
```
Comment (airnet73) [inline src/executables/csql.c]: stmts may be larger than 100 bytes.

Comment (eido5) [reply src/executables/csql.c]: Right. Its size expands till the memory lack.

I should consider about it. Thank you.

File: src/executables/csql.c [hunk 4]
```diff
@@ -1756,6 +1790,8 @@ csql_execute_statements (const CSQL_ARGUMENT * csql_arg, int type, const void *s
   bool do_abort_transaction = false;	/* flag for transaction abort */
   PT_NODE *statement = NULL;
   char sql_text[DDL_LOG_BUFFER_SIZE] = { 0 };
+  char prompt_cpy[100];		/*static char csql_Prompt[100] */
```
Comment (eido5) [inline src/executables/csql.c]: necessary to fix

File: src/executables/csql.c (hunk 3 above)
Comment (eido5) [reply src/executables/csql.c]: add_history is needed to be fixed.

**Pull request closed without merging**